tools/dynamo/gb_id_mapping.py

```diff
@@ -0,0 +1,146 @@
+
+    return expanded_hints
+
+
+def find_unimplemented_v2_calls():
+    dynamo_dir = Path(__file__).parent.parent.parent / "torch" / "_dynamo"
+    results = []
+
+    for file_path in dynamo_dir.glob("**/*.py"):
+        with open(file_path, "r") as f:
+            source = f.read()
+            try:
+                tree = ast.parse(source)
+
+                for node in ast.walk(tree):
+                    if isinstance(node, ast.Call) and hasattr(node, "func"):
+                        is_unimplemented = False
+                        if (
+                            isinstance(node.func, ast.Name)
+                            and node.func.id == "unimplemented_v2"
+                        ):
+                            is_unimplemented = True
+                        elif (
+                            isinstance(node.func, ast.Attribute)
+                            and node.func.attr == "unimplemented_v2"
+                        ):
+                            is_unimplemented = True
+
+                        if is_unimplemented:
+                            info = {
+                                "gb_type": None,
+                                "context": None,
+                                "explanation": None,
+                                "hints": [],
+                                "from_exc": None,
+                            }
+
+                            for kw in node.keywords:
+                                if kw.arg in info:
+                                    param_source = get_source_segment(source, kw.value)
+                                    if param_source:
+                                        info[kw.arg] = clean_string(param_source)
+                                    elif isinstance(kw.value, ast.Constant):
+                                        info[kw.arg] = kw.value.value
+                                    else:
+                                        info[kw.arg] = "DYNAMIC_CONTEXT"
+
+                            results.append((file_path, node.lineno, info))
+            except SyntaxError:
+                print(f"Syntax error in {file_path}")
+
+    return results
+
+
+def create_registry():
+    calls = find_unimplemented_v2_calls()
+    registry = {}
+
+    gb_types = {}
+    for file_path, line, info in calls:
+        if info["gb_type"] and info["gb_type"] not in gb_types:
+            gb_types[info["gb_type"]] = info
+
+    for i, (gb_type, info) in enumerate(sorted(gb_types.items()), 1001):
+        gb_id = f"GB{i}"
+
+        hints = info["hints"]
+        if isinstance(hints, str):
+            if hints.startswith("[") and hints.endswith("]"):
+                try:
+                    items = re.findall(r'"([^"]*)"', hints)
+                    if items:
+                        hints = items
+                    elif "*graph_break_hints." in hints:
+                        hints = expand_hints([hints])
+                except:
+                    pass
```

Comment:
This makes more sense from an organization point of view to be placed in find_unimplemented_v2_calls